packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1870,6 +1870,11 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     if (!decoration.enabled)
       return themeData.disabledColor;
 
+      // Checks if unselectedWidgetColor property is not one of the defaults and returns based on the result
+    if (themeData.unselectedWidgetColor != Colors.black54??Colors.white70){
```

Comment:
This probably isn't what you want, see http://blog.sethladd.com/2015/07/null-aware-operators-in-dart.html

InputDecorator uses `IconTheme` to define the default icon colors for widgets like prefixIcon and suffixIcon. Applications can override the default by creating a widget that includes an Icon that specifies the [icon's color (https://api.flutter.dev/flutter/widgets/Icon/color.html).

Reply:
I love learning from my mistakes so the more the merrier.

I changed the code as per your spec however I noticed that we get the focused color from seperate sources and not Icon Theme. Should I change it for consistency or should we best leave as is?